src/components/featureHighlight/index.js

```diff
@@ -148,11 +148,27 @@ class FeatureHighlight extends BaseComponent {
     this.setTargetPosition();
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    this.setTargetPosition(nextProps);
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const target = nextProps?.getTarget?.();
+    const node = FeatureHighlight.findTargetNode(target);
+    if (node && node !== prevState?.node) {
+      return {...prevState, node};
+    }
+    return null;
   }
 
-  componentDidUpdate() {
+  shouldSetTargetPosition = (nextProps) => {
```

Comment:
Looks like 'shouldComponentUpdate'... If it's not a pure component you can just implement that method instead and the component will not be updated if it doesn't meet the condition.

Reply:
I've tried it before and now again, it's not quite working.
the position of the `FeatureHighlight` view is getting the previous data, that's why I went with this solution.
(I even tried `!_.isEqual(nextProps, this.props)`)

Reply:
Wierd. Ok...